openmdao/core/problem.py

```diff
@@ -408,23 +408,31 @@ def __setitem__(self, name, value):
         """
         self.set_val(name, value)
 
-    def set_val(self, name, value, units=None, indices=None):
+    def set_val(self, name, val, units=None, indices=None, value=None):
         """
         Set an output/input variable.
 
-        Function is used if you want to set a value using a different unit.
+        Function is used if you want to set a val using a different unit.
 
         Parameters
         ----------
         name : str
             Promoted or relative variable name in the root system's namespace.
-        value : float or ndarray or list
-            Value to set this variable to.
+        val : float or ndarray or list
+            Val to set this variable to.
         units : str, optional
-            Units that value is defined in.
+            Units that val is defined in.
         indices : int or list of ints or tuple of ints or int ndarray or Iterable or None, optional
-            Indices or slice to set to specified value.
+            Indices or slice to set to specified val.
+        value : float or ndarray or list or None
+            This argument is deprecated. Use 'val' instead.
         """
+        if val is None and value is None:
+            raise RuntimeError("Val is required.")
+        elif value is not None:
+            val = value
+            warn_deprecation("'value' will be deprecated in 4.0. Please use 'val' in the future.")
```

Comment:
I think you need some additional info in this warning message to point the user to the call in question
And I think you mean to say that the argument *is* deprecated and will be *removed* in 4.0?
e.g.:
```
warn_deprecation(f"{self.msginfo} when calling `set_val` on {name}, the 'value' argument is deprecated and will be removed in version 4.0. Please use 'val'.")
```